Assets/Scripts/NPC/WalkingState.cs

```diff
@@ -0,0 +1,82 @@
+using UnityEngine;
+
+public class WalkingState : SmallEnemiesBaseState
+{
+    [SerializeField] private Waypoint currentWaypoint;
+    [SerializeField] private Waypoint lastWaypoint;
+    [SerializeField] private Grid myGrid;
+    [SerializeField] private float distanceTreshhold;
+    [SerializeField] private float movingSpeed;
+
+    [Header("Performance")]
+    [SerializeField] private int maxWaypointCallStack = 2;
+
+    private int newWaypointCallStack;
+    private Rigidbody2D _rigidbody;
```

Comment:
Je zou deze refence kunnen verplaatsen naar de enemy walking state machine, zodat andere state in de toekomst ook de zelfde refrence kunnen gebruiken.